### PR TITLE
Move worn climate control to a JSON flag

### PR DESCRIPTION
**Pull request by darktoes**
First of all, props to plazma-rush for his recent PR which showed me how to actually do this.
Also, I'm not 100% sure the code I wrote will actually work, so if someone could check it that would be awesome.

**Problem**
Currently, worn climate control is handled by checking whether the player is wearing activated power armor or activated RM13 Combat Armor. This means that it's impossible to make climate controlled clothing through JSON, limiting modders and flexibility.

**Intentions/Features**
This PR changes the code so that climate control is achieved just by having a piece of clothing with the "CCONTROL" JSON flag on it, as well as adding that tag to RM13 armor so that it doesn't lose it's climate control.

**Implementation**
Quite simple really. Player.cpp is changed so that the in_climate_control( ) value returns true if the player is wearing an item with the "CCONTROL" flag, making it far easier to make items with climate control, even allowing modders to make items with it.

For tidiness, RM13 armor is removed from the in_climate_control( ) check and instead is given the "CCONTROL" tag, so it retains it's climate control ability.

Adds the new tag to the flags.JSON and to the documentation so people can use it.

### CODE REVIEW

**Comment by darktoes** [comment]
Not sure why it kept the bit about updating my master. Seems I did something weird.

**Comment by kevingranade** [comment]
Please change the flag to CLIMATE_CONTROL
We don't pay by the letter ;)

File: data/json/flags.json [hunk 1]
```diff
@@ -50,6 +50,12 @@
     "context": [ "BIONIC_ITEM" ],
     "info": "This item is a Compact Bionic Module.  You can <info>use an autodoc</info> or <info>ask a surgeon</info> to install it into your body."
   },
+  {
+    "id": "CCONTROL",
+    "type": "json_flag",
+    "context": [ "ARMOR", "TOOL_ARMOR" ],
+    "info": "This piece of clothing has <info>built-in or attached climate control,</info> keeping you at a <good>comfortable temperature.</good>"
```
Comment (Night-Pryanik) [inline data/json/flags.json]: Move comma outside of tags.

**Comment by darktoes** [comment]
Ugh. You're lucky you're cute, Kevin.

File: data/json/items/tool_armor.json [hunk 2]
```diff
@@ -560,7 +560,7 @@
         "name": "RM13 combat armor (on)",
         "name_plural": "RM13 combat armors (on)",
         "description": "Internally powered by a maximum of ten plutonium fuel cells, this full-body suit of sleek black military armor represents the pinnacle of Rivtech's non-rigid powered armor technology.  It is turned on, and continually draining power.  Use it to turn it off.",
-        "flags" : ["VARSIZE", "STURDY", "WATERPROOF", "RAINPROOF", "WATCH", "ALARMCLOCK", "SWIM_GOGGLES", "SUN_GLASSES", "RAD_PROOF", "ELECTRIC_IMMUNE", "THERMOMETER", "TRADER_AVOID"],
+        "flags" : ["VARSIZE", "STURDY", "WATERPROOF", "RAINPROOF", "WATCH", "ALARMCLOCK", "SWIM_GOGGLES", "SUN_GLASSES", "RAD_PROOF", "ELECTRIC_IMMUNE", "THERMOMETER", "TRADER_AVOID", "CCONTROL"],
```
Comment (ZhilkinSerg) [inline data/json/items/tool_armor.json]: Flag was not renamed here.

File: src/player.cpp [hunk 3]
```diff
@@ -2579,10 +2579,10 @@ bool player::in_climate_control()
         return true;
     }
     for( auto &w : worn ) {
-        if( w.typeId() == "rm13_armor_on" ) {
+        if( w.active && w.is_power_armor() ) {
             return true;
         }
-        if( w.active && w.is_power_armor() ) {
+        if( worn_with_flag( "CCONTROL" ) ) {
```
Comment (ZhilkinSerg) [inline src/player.cpp]: Flag was not renamed here too.

File: data/json/flags.json (hunk 1 above)
Comment (darktoes) [reply data/json/flags.json]: Fixed.

File: src/player.cpp (hunk 3 above)
Comment (darktoes) [reply src/player.cpp]: You reviewed it too fast...

File: data/json/items/tool_armor.json (hunk 2 above)
Comment (darktoes) [reply data/json/items/tool_armor.json]: Damnit. Fixed.

**Pull request merged**